src/scanner.ts

```diff
@@ -21,9 +62,10 @@ export interface InclusiveDiagnostic {
  */
 export async function scanFile(filePath: string): Promise<InclusiveDiagnostic[]> {
     // read entire file is faster than going through lines
-    let fileContent: string = readFileSync(filePath, 'utf8');
+    let fileInfo = new FileInfo(filePath);
```

Comment:
this structure is way better than passing the file content as string. Great job @edguer! 👍